### ISSUE TITLE
Remove unnecessary calls to `await Task.CompletedTask;`

### ISSUE THREAD

**Issue by mrlacey** [Microsoft/WindowsTemplateStudio#2068]
Many templates include `await Task.CompletedTask;` in the default generated code so that the code compiles when generated and we don't need to add the `async` modifier to the method definition when we merge calls into the method body as pages and features are added to the project.
An example of this is in `ActivationService.StartupAsync`.

Once these methods have genuine async calls added, the call to `await Task.CompletedTask` becomes unnecessary and can (should) be deleted. 
We already do a similar thing where we remove calls to `yield break` when it's no longer needed. It would be nice to increase the consistency in how we treat generated code and not include code in the generated project that is not required.

There are also some PRISM templates that `return Task.CompletedTask;` rather than awaiting it. This means that method signatures don't match what might be expected. There are also some calls to methods with names ending 'Async' which aren't awaited. This breaks expectations on naming and calling conventions. - If reviewing/addressing this turns into a lot of work, consider splitting into a separate issue.

**Comment by sibille** [comment]
We had a look at this and detected another issue with deleting code on mergepostactions that blocks this issue (x-ref: #2088)
Apart from that I'm not sure if it will be possible to remove `await Task.CompletedTask;` as we rely on this code to insert code at the end of `ActivationService.StartupAsync` to solve issue #1752

**Comment by mrlacey** [comment]
I thought the changes made previously allowed things to get inserted at the top by default and the `//^^`  symbol was used to ensure something went at the bottom.

**Comment by sibille** [comment]
Yes, thats true. if you put //^^ is inserted above the next reference line that is found. 
So the postaction for FirstRunDialog for example is the following. 
```
private async Task StartupAsync()
{
    //^^
    //{[{ 
    await FirstRunDisplayService.ShowIfAppropriateAsync(); 
    //}]} 
    await Task.CompletedTask; 
} 
```
If the line await Task.CompletedTask; is removed I think this will fail. I'm not 100% sure, but wanted to add the observation to make sure that the person that implements this change is aware and can look for a solution to both issues, we might be able to remove the await CompletedTask with a globalpostaction maybe.

**Comment by sibille** [comment]
Movin to 2.2 as this is not high priority and depends on 2088

**Comment by mvegaca** [comment]
Review in dev center notification
![image](https://user-images.githubusercontent.com/5556784/40832133-0c8ff462-658b-11e8-93b2-518ae80b735f.png)
